packages/hooks/src/useHandleEscape.ts

```diff
@@ -0,0 +1,25 @@
+import useEventListener, { UseEventOptions } from './useEventListener';
+
+const escapeKeyCode = 27;
+
+const handleEscape = (e: KeyboardEvent, callback: () => void) => {
+  if (e.keyCode === escapeKeyCode) {
+    e.stopImmediatePropagation();
```

Comment:
Why do we want this? This will stop any other keydown listeners from firing, curious why we want this?

Reply:
Perhaps it should be handled on a per usage basis instead of globally?

I guess I'm looking for the thinking behind this as removing it may be tricky.

Reply:
We always had `e.stopImmediateProp()` when using the generic version in Modal, Menu and Tooltip components. We can definitely think about removing or finding a way to handle per usage?

Reply:
Perhaps we should call [stopPropagation](https://stackoverflow.com/a/5299841) instead? ImmediatePropagation seems a little heavy and could conflict with other application code.

Reply:
Updates to useEscape: https://github.com/mongodb/leafygreen-ui/pull/168